src/main/cpp/subsystems/Claw.cpp

```diff
@@ -1,27 +1,116 @@
 #include "subsystems/Claw.h"
+#include "Parameters.h"
+#include "Utils.h"
+#include "ui/UserInterface.h"
+
+bool Claw::Init() {
+    // read PID values from the parameters file
+    double kP = RobotParams::GetParam("claw.claw_pid.kP", 0.01);
+    double kI = RobotParams::GetParam("claw.claw_pid.kI", 0.0);
+    double kD = RobotParams::GetParam("claw.claw_pid.kD", 0.0);
+
+    // create the claw PID
+    claw_pid_ = std::make_shared<frc::PIDController>(kP, kI, kD);
+
+    // manual control
+    claw_power_ = 0;
+
+    mode_ = Manual;
+
+    // logs for the claw values
+    claw_output_log_ = wpi::log::DoubleLogEntry(TeamOKC::log, "/claw/claw_output");
+    claw_enc_log_ = wpi::log::DoubleLogEntry(TeamOKC::log, "/claw/claw_enc");
+    claw_setpoint_log_ = wpi::log::DoubleLogEntry(TeamOKC::log, "/claw/claw_setpoint");
+
+    // preset positions from parameters.toml
+    open_pos_ = RobotParams::GetParam("claw.open", 0);
+    cube_pos_ = RobotParams::GetParam("claw.cube", 0);
+    cone_pos_ = RobotParams::GetParam("claw.cone", 0);
+
+    this->Reset();
 
-bool Claw::Init(){
     return true;
 }
 
-bool Claw::ResetPositionEncoder(){
+bool Claw::ResetPositionEncoder() {
+    interface_->reset_claw_open_and_close = true;
+
     return true;
 }
-bool Claw::ResetPositionPID(){
+
+bool Claw::ResetPositionPID() {
+    claw_pid_->Reset();
+
     return true;
 }
-bool Claw::SetPosition(){
-    // SetPosition (30); //cone
-    // SetPosition (0); //close
-    // SetPosition (50);//cube
+
+bool Claw::SetPreset(ClawPreset preset) {
+    // set the claw to a particular preset
+
+    if (preset == OPEN) {
+        this->SetPosition(open_pos_);
+    } else if (preset == CUBE) {
+        this->SetPosition(cube_pos_);
+    } else if (preset == CONE) {
+        this->SetPosition(cone_pos_);
+    } else {
+        // we shouldn't have gotten here, so let someone know something weird happened
+        OKC_CHECK_MSG(false, "unhandled claw enum value");
```

Comment:
maybe mention "preset" in here somewhere to avoid confusion with the Manual/Auto enum

Reply:
good idea